shared/people/invite-friends/invite-contacts.native.tsx

```diff
@@ -0,0 +1,174 @@
+  contacts
+    .filter(contact => contact.name.toLowerCase().includes(search) || contact.value.includes(search))
+    .forEach(contact => {
+      const category = categorize(contact)
+      const section = mapGetEnsureValue(sectionMap, category, [])
+      section.push(contact)
+    })
+  const sections: Section[] = []
+  for (const letter of 'ABCDEFGHIJKLMNOPQRSTUVWXYZ') {
+    if (sectionMap.has(letter)) {
+      sections.push({
+        data: sectionMap.get(letter)!,
+        key: letter,
+        title: letter,
+      })
+    }
+  }
+  for (const sectionKey of ['0-9', 'Other']) {
+    if (sectionMap.has(sectionKey)) {
+      sections.push({
+        data: sectionMap.get(sectionKey)!,
+        key: sectionKey,
+        title: sectionKey,
+      })
+    }
+  }
+  return sections
+})
+
+const InviteContacts = () => {
+  const {contacts, errorMessage, loading} = useContacts()
+  const [search, setSearch] = React.useState('')
+  const [selectedContacts, setSelectedContacts] = React.useState(new Set<string>())
+  const canContinue = selectedContacts.size > 0
+
+  const dispatch = Container.useDispatch()
+  const nav = Container.useSafeNavigation()
+  const navUp = () => dispatch(nav.safeNavigateUpPayload())
+  const onSubmit = () => {} // TODO
+
+  const placeholderText = loading ? '' : `Search ${contacts.length.toLocaleString()} contacts`
+
+  const sections = [
+    {data: [], key: 'headerIcon', title: 'headerIcon'},
+    ...filterAndSectionContacts(contacts, search),
+  ]
+  const renderSectionHeader = ({section}: {section: Section}) =>
+    section.title === 'headerIcon' ? (
+      <Kb.Icon type="icon-illustration-invite-friends-460-96" style={styles.iconBox} />
+    ) : (
+      <Kb.SectionDivider label={section.title} />
+    )
+  const renderItem = ({item, index}: {item: Contact; index: number}) => {
+    const topText = item.name ?? item.valueFormatted ?? item.value
+    const bottomText = item.name ? item.valueFormatted ?? item.value : undefined
+    const onCheck = (check: boolean) => {
+      if (check) {
+        selectedContacts.add(item.value)
+      } else {
+        selectedContacts.delete(item.value)
+      }
+      setSelectedContacts(new Set(selectedContacts))
+    }
+    const checked = selectedContacts.has(item.value)
+    return (
+      <Kb.ListItem2
+        type="Small"
+        firstItem={index === 0}
+        body={
+          <Kb.Box2 direction="vertical" alignItems="flex-start">
+            <Kb.Text type="BodySemibold">{topText}</Kb.Text>
+            {bottomText && <Kb.Text type="BodySmall">{bottomText}</Kb.Text>}
+          </Kb.Box2>
+        }
+        action={<Kb.CheckCircle checked={checked} onCheck={onCheck} style={styles.checkCircle} />}
+        icon={
+          item.pictureUri ? (
+            <Kb.NativeImage style={styles.thumbnail} source={{uri: item.pictureUri}} />
+          ) : (
+            undefined
+          )
+        }
+      />
+    )
+  }
+  const keyExtractor = (item: Contact) => item.value
```

Comment:
This is fragile, can we use something more likely to be unique? IIRC native contacts plumb over some kind of ID, maybe that can be used?